mongo/client.go

```diff
@@ -1043,6 +1045,7 @@ func (c *Client) UseSession(ctx context.Context, fn func(SessionContext) error)
 }
 
 // UseSessionWithOptions operates like UseSession but uses the given SessionOptions to create the Session.
+// UseSessionWithOptions is not thread safe or fork safe.
```

Comment:
Similar to above: `UseSession` is actually safe to call from multiple goroutines, but the the `SessionContext` passed to the callback func is not. Also, prefer Go-specific terminology for describing concurrency to prevent confusion.